src/jit/objectalloc.cpp

```diff
@@ -246,6 +258,63 @@ void ObjectAllocator::ComputeEscapingNodes(BitVecTraits* bitVecTraits, BitVec& e
     }
 }
 
+//------------------------------------------------------------------------------
+// ComputeStackObjectPointers : Given an initial set of possibly stack-pointing nodes,
+//                              and an initial set of definitely stack-pointing nodes,
+//                              update both sets by computing nodes reachable from the
+//                              given set in the reverse connection graph.
+//
+// Arguments:
+//    bitVecTraits                    - Bit vector traits
+
+void ObjectAllocator::ComputeStackObjectPointers(BitVecTraits* bitVecTraits)
+{
+    bool changed = true;
+
+    while (changed)
+    {
+        changed = false;
+        for (unsigned int lclNum = 0; lclNum < comp->lvaCount; ++lclNum)
+        {
+            LclVarDsc* lclVarDsc = comp->lvaTable + lclNum;
+            var_types  type      = lclVarDsc->TypeGet();
+
+            if (type == TYP_REF || type == TYP_I_IMPL || type == TYP_BYREF)
+            {
+                if (!MayLclVarPointToStack(lclNum) &&
+                    !BitVecOps::IsEmptyIntersection(bitVecTraits, m_PossiblyStackPointingPointers,
+                                                    m_ConnGraphAdjacencyMatrix[lclNum]))
+                {
+                    // We discovered a new pointer that may point to the stack.
+                    MarkLclVarAsPossiblyStackPointing(lclNum);
+
+                    // Check if this pointer always points to the stack.
+                    if (lclVarDsc->lvSingleDef == 1)
```

Comment:
We should really get around to writing a checker for the early meaning of `lvSingleDef` someday. I think the upstream phases maintain it, but ...